### ISSUE TITLE
How are Suggestions and Comments matched with a given translation string?

### ISSUE THREAD

**Issue by akien-mga** [WeblateOrg/weblate#6988]
### Describe the issue

Hi,

In a (Hosted) Weblate repository with gettext PO files ([godot-engine/godot-docs](https://hosted.weblate.org/projects/godot-engine/godot-docs), I recently moved the POT and PO files in Git (moving from `weblate/docs/*.po*` to `weblate/*.po*`), and then Reset the Weblate repo to track the upstream one (https://github.com/godotengine/godot-docs-l10n).

I then modified the Weblate project settings to match the new paths and it properly reimported all the translations.

Now I notice that the Suggestions and Comments columns are all empty, which in retrospective is something I should have thought about. I decided to open this support ticket to clarify how Suggestions and Comments are stored by Weblate and matched to a given msgid, so that users can know which operations may result in loss of data.

My hunch is that it might be both file and msgid based?
If so, would moving my files back to their original location restore the Suggestions and Comments that had been issued prior to that move?

For the record, I don't know if there were any Suggestions and Comments, but I'm fairly sure there would have been. It's a huge translation catalog with dozens of target languages, and other Godot resources do have a fair number of Suggestions so I assume it was similar here.

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

- Create a Weblate project tracking a Git repo with gettext POT and PO files
- Do some translations on Weblate, add suggestions to some resources, commit
- In the Git repo, `git mv` the POT and PO files, commit
- Reset the Weblate repo, change settings to track the new POT and PO locations
- See that after sync of the translations, the suggestions are no longer listed

### Expected behavior

From a technical point of view I think the current behavior *is* expected. But clarifying this in the documentation, and possibly in the interface, could be useful. Or simply just having it documented in this issue for future users who might do the same mistake.

### Screenshots

See how the Suggestions and Comments columns don't show any entry after the above described destructive operation.
I do note that the English source still has its comments though, even though the POT moved too. I assume Weblate keeps track of source comments differently from translation comments.

![Screenshot_20211222_110854](https://user-images.githubusercontent.com/4701338/147075710-6e311faf-ef8d-47f6-8841-be3f7c10a48e.png)

Here's another Godot repo where things haven't been moved around and most languages have suggestions and/or comments:
![image](https://user-images.githubusercontent.com/4701338/147075959-a881e8af-efc1-496b-b8fb-acb18ba648f8.png)

### How do you run Weblate?

weblate.org service

### Weblate versions

Weblate 4.10, hosted on weblate.org

**Comment by nijel** [comment]
In the database, these are linked as foreign key to the string. So if you delete the string, the comments and suggestions are deleted as well. We used to have content based linking years ago, but that lead to numerous issues in the codebase.

What you did ended up in deleting all existing strings and creating them again, what did cause loss of the additional content.

What could you do to avoid that (the two `git mv` and reset Weblate steps):

1. Lock Weblate component
2. Commit pending changes from Weblate and merge them
3. Disable receiving webhooks the Weblate project configuration; this avoids Weblate immediatelly seeing changes in the repo
4. Do any needed changes in the repo (`git mv`), push them
5. Change component configuration to match new setup; upon changing configuration, Weblate will fetch the updated repository and notice the changed locations while keeping existing strings
6. Unlock the component and re-enable hooks in the project configuration

I know this is not really straightforward, but currently I don't see a better solution.

**Comment by akien-mga** [comment]
Thanks, that definitely clarifies what happens.

In my update process I mistakenly reset the Weblate repo *before* changing the component configuration to match the new setup, for a few hours we did end up with only the pre-move source strings being found in the component. And after I fixed the configuration it did re-parse and re-add all translated strings.

So it's indeed pretty obvious that I mistakenly deleted and re-created all content.

I think the issue can be closed then, it should provide useful context for future users running into the same issue.
